ansible/edpm_provisioning.yml

```diff
@@ -0,0 +1,204 @@
+---
+- name: Provisioning EDPM nodes
+  hosts: localhost
+  gather_facts: true
+
+  tasks:
+    - name: Download OCP inventory file (scalelab)
+      uri:
+        url: "https://wiki.scalelab.redhat.com/instack/{{ cloud }}_ocpinventory.json"
+        return_content: true
+      register: ocpinventory_scale
+      when: lab == 'scalelab'
+
+    - name: Download OCP inventory file (performancelab)
+      uri:
+        url: "https://wiki.rdu3.labs.perfscale.redhat.com/instack/{{ cloud }}_ocpinventory.json"
+        return_content: true
+      register: ocpinventory_alias
+      when: lab == 'performancelab'
+
+    - name: Set fact for OCP inventory
+      set_fact:
+        ocp_inventry: "{{ ocpinventory_scale if lab == 'scalelab' else ocpinventory_alias }}"
+
+    - name: Parse nodes from OCP inventory
+      set_fact:
+        all_nodes: "{{ (ocp_inventry.content | from_json).nodes }}"
+
+    - name: Set final compute count
+      set_fact:
+        final_compute_count: >-
+          {% set nodes_length = ocp_inventry.json.nodes | length %}
+          {% if compute_count == -1 %}
```

Comment:
currently, we dont have to consider the case compute_count == -1.
We can provision rest of the nodes excluding ocp + bastion.
if ceph is enabled(this can be determined by ceph_backend), automation should avoid provisioning ceph_node_count from the bottom of the inventory.
- when someone deploying actual external ceph, they should be able to reuse this role for provisioning ceph nodes.